net/Socket.hpp

```diff
@@ -362,26 +410,47 @@ class Socket
     /// Construct based on an existing socket fd.
     /// Used by accept() only.
     Socket(const int fd, Type type)
-        : _fd(fd)
+        : _type(type),
+          _clientPort(0),
+          _fd(fd),
+          _open(_fd >= 0),
+          _creationTime(std::chrono::steady_clock::now()),
```

Comment:
now() has some cost - we should where possible reduce its usage and pass around the 'now' from the poll that accepted the socket etc.

I'm somewhat nervous that we initialize things here rather than in init. There are two socket constructors, and using 'init' is (cf. the comment above) a way of not duplicating code & state setup between them. Please move as much shared code as possible into init; we could pass 'type' and 'fd' as parameters to init too here nicely.

Reply:
Aware of the stead monotonic clock now() costs, AFAIK the caller does not have it at this event (but will double check).
- _type is now const
- _creationTime was added as const
- _fd was always const (wan't my idea, but a good one)

The immutable nature clarifies semantics here and well possible if no assignment operator is used/allowed.
Sure, we could live w/o immutable semantics here - as long we have 'em private and use accessor methods as we have now. So .. well, either way. Still, having fields immutable is a good helper :)
Note: Compiler may behave differently .. but that is my least criteria for this code here.
Bottom line: I would stick with the current code as no algorithmic duplication really exist, but will check whether caller of ctor has now() already fetched - if that is OK.

Reply:
very common failure mode is un-initialized members; the more we share there the safer we are. Lets share all we can =) The 'const' stuff is fine - I guess until we get proper chained constructors in C++.

Pushing 'now' around the code to more places is something that happens a fair bit - and is easy to special-case if it's not needed - just have a default argument for it; and then tweak it in the places that matter I guess. Still sockets are not created that often so prolly not a big deal.

Reply:
added ctor arg w/ default, was a PIA and probably not impacting really - but another good reason to walk around the code.